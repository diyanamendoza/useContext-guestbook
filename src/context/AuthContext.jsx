import { createContext, useState, useContext } from "react";

const AuthContext = createContext()

const AuthProvider = ({ children }) => {
    const [user, setUser] = useState('')
    const login = (username, password) => {
        const loginOk = username === process.env.AUTH_USER && password === process.env.AUTH_PASS
        if (loginOk) setUser({ username })
        return loginOk
    }
    const logout = () => {
        setUser(null)
    }

    return (
        <AuthContext.Provider value={{ user, login, logout }}> {children} </AuthContext.Provider>
    )
}

const useAuth = () => {
    const context = useContext(AuthContext)
    if (context === undefined) {
        throw new Error('useAuth not used inside provider')
    }
    return context
}

export { AuthProvider, useAuth }