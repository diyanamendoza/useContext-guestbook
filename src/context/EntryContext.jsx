import { useContext, createContext, useState } from 'react'

const EntryContext = createContext()

const EntryProvider = ({ children }) => {
    const [entry, setEntries] = useState([])
    return <EntryContext.Provider value={{ entry, setEntries }}>{children}</EntryContext.Provider>
}

const useEntries = () => {
    const context = useContext(EntryContext)
    if (context === undefined) {
        throw new Error('The useEntries hook is not being called inside a EntryContext.Provider')
    }
    return context
}

export { EntryProvider, useEntries }