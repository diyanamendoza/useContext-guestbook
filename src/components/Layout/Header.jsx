import { useUser } from "../../context/UserContext"

export default function Header() {
    const { user } = useUser()

    return (
        <header>
            { user ? <p>Signing as {user}</p> : <p>Sign In Below</p>}
        </header>
    )
}
