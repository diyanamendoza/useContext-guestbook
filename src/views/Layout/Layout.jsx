import Header from "../../components/Layout/Header";

export default function Layout({ children }) {
    return (
        <>
           <Header/>
           <main>{children}</main>
        </>
    )
}
