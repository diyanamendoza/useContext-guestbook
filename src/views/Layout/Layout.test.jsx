import { render } from '@testing-library/react'
import Layout from './Layout'
import { EntryProvider } from '../../context/EntryContext'
import { UserProvider } from '../../context/UserContext'

it('should render Layout', () => {
  const { container } = render(
    <EntryProvider>
    <UserProvider>
        <Layout />
    </UserProvider>
    </EntryProvider>
  )
  expect(container).toMatchSnapshot()
})