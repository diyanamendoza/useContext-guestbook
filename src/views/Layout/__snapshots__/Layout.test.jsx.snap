// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Layout 1`] = `
<div>
  <header>
    <p>
      Sign In Below
    </p>
  </header>
  <main />
</div>
`;
